server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -157,6 +189,10 @@ public void registerHandler(RestRequest.Method method, String path, RestHandler
 
     @Override
     public void dispatchRequest(RestRequest request, RestChannel channel, ThreadContext threadContext) {
+        if (tracerLog.isTraceEnabled() && shouldTraceRequest(request.uri())) {
+            tracerLog.trace(new ParameterizedMessage(
+                "Incoming request [{}][{}] on [{}]", request.method(), request.uri(), request.getHttpChannel()));
```

Comment:
Should we use the same syntax as for transport-level logging, i.e., `[{}][{}] received request`?

Reply:
I think we can align them more closely yes if we're also going to log the response sending now. On it :)

Reply:
How about:

```
[2020-01-31T10:09:34,141][TRACE][o.e.h.A.tracer           ] [testTracerLog] [43072717][OPTIONS][/internal/test] received request from [org.elasticsearch.test.rest.FakeRestRequest$FakeHttpChannel@2402f9fe]
[2020-01-31T10:09:34,153][TRACE][o.e.h.A.tracer           ] [testTracerLog] [43072717] sent response to [org.elasticsearch.test.rest.FakeRestRequest$FakeHttpChannel@2402f9fe]

```
?

I used the hash code of the request to make a fake but good enough request id for each REST request.